src/components/ShareButtons/ShareButtonGroup.tsx

```diff
@@ -10,36 +10,60 @@ import CopyLinkButton from './CopyLinkButton';
 import { ShareButton, SocialButtonsContainer } from './ShareButtons.style';
 
 const ShareImageButtons: React.FC<{
-  imageUrl: string;
+  imageUrl: string | (() => Promise<string>);
   imageFilename: string;
-  url: string;
+  url: string | (() => Promise<string>);
```

Comment:
I had to rework this so that url and imageUrl can be created lazily and asynchronously (we don't generate them until you actually click the share / save buttons).  So you can either pass a string or a function that runs and returns a `Promise<string>` now...